extensions/sql-database-projects/src/templates/templates.ts

```diff
@@ -4,27 +4,85 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as path from 'path';
+import * as constants from '../common/constants';
 import { promises as fs } from 'fs';
 
-// Project templates
 export let newSqlProjectTemplate: string;
 
-// Script templates
+// Object types
 
-export let newSqlScriptTemplate: string;
-export let newSqlTableTemplate: string;
-export let newSqlViewTemplate: string;
-export let newSqlStoredProcedureTemplate: string;
+export const script: string = 'script';
+export const table: string = 'table';
+export const view: string = 'view';
+export const storedProcedure: string = 'storedProcedure';
+export const folder: string = 'folder';
+
+// Object maps
+
+let scriptTypeMap: Record<string, ProjectScriptType> = {};
+
+export function projectScriptTypeMap(): Record<string, ProjectScriptType> {
+	if (Object.keys(scriptTypeMap).length === 0) {
+		throw new Error('Templates must be loaded from file before attempting to use.');
+	}
+
+	return scriptTypeMap;
+}
+
+let scriptTypes: ProjectScriptType[] = [];
+
+export function projectScriptTypes(): ProjectScriptType[] {
+	if (scriptTypes.length === 0) {
+		throw new Error('Templates must be loaded from file before attempting to use.');
+	}
+
+	return scriptTypes;
+}
 
 export async function loadTemplates(templateFolderPath: string) {
+
 	newSqlProjectTemplate = await loadTemplate(templateFolderPath, 'newSqlProjectTemplate.xml');
 
-	newSqlScriptTemplate = await loadTemplate(templateFolderPath, 'newTsqlScriptTemplate.sql');
-	newSqlTableTemplate = await loadTemplate(templateFolderPath, 'newTsqlTableTemplate.sql');
-	newSqlViewTemplate = await loadTemplate(templateFolderPath, 'newTsqlViewTemplate.sql');
-	newSqlStoredProcedureTemplate = await loadTemplate(templateFolderPath, 'newTsqlStoredProcedureTemplate.sql');
+	await loadObjectTypeInfo(script, constants.scriptFriendlyName, templateFolderPath, 'newTsqlScriptTemplate.sql');
+	await loadObjectTypeInfo(table, constants.tableFriendlyName, templateFolderPath, 'newTsqlTableTemplate.sql');
```

Comment:
is it possible to load all these as parallel?

Reply:
Good idea.  Yeah, they can be.